cli/server.go

```diff
@@ -1945,6 +1945,13 @@ func redirectToAccessURL(handler http.Handler, accessURL *url.URL, tunnel bool,
 			return
 		}
 
+		for _, ignore := range ignoreHosts {
+			if r.Host == ignore {
+				handler.ServeHTTP(w, r)
+				return
+			}
+		}
```

Comment:
This ain't gonna work.  We install this middleware at start of day, before we learn about any other replicas, and they change over time.

I think we should blanket refuse to redirect the path `/derp`.  DERP is encrypted, so even if some client connection got the http URL for this path, it would be fine.

Reply:
How so? We're only looking at the redirect for ourself, so it doesn't matter which replicas come up or down, it'll always be the same host locally.

Reply:
No, you're right.  I'm thinking about it the wrong way around. Sorry.

However, disabling the redirect for _all routes_ on that host feels too broad.  I think we generally do want to redirect for non-DERP routes if the request comes to e.g. our IP address or a non-cannonical hostname.